### ISSUE TITLE
Aspiring software developer seeking advice

### ISSUE THREAD

**Issue by nicholasrokosz** [spences10/ama#13]
Hi Scott,
I'm glad to have discovered your website last week. Your posts on Gatsby and MDX are super helpful.

Brief overview of my situation:
- I'm a college dropout with few good references and an unimpressive resume
- I taught myself programming at the beginning of the pandemic by reading beginner python books
- I enrolled in a web dev bootcamp to learn more and interact with other aspiring software devs

The bootcamp ends in a week and I'm very nervous about the job hunt. I'm actually confident that I could succeed in a junior developer role but unsure of the best way to present myself to potential employers.

What would you like to see from someone in my position in order to get a first job in software development?

and a followup: What advice do you have for someone in a junior dev role who would like to eventually be in the position to work at interesting companies (such as Karmarama)?

Any advice is greatly appreciated.
Thanks,
Nick Rokosz

**Comment by spences10** [comment]
Hey @nicholasrokosz

Thanks for reaching out 🙏 I posted your stickers you won this
afternoon 😊 thanks for subscribing to [my newsletter].

## Tl;Dr

Have a visible profile, so if I wanted to check you out I can see your
GitHub and some projects you've worked on. (This is what I did when
seeing you post this 👍)

If the projects on your GitHub profile are sites then have them hosted
somewhere (Vercel/Netlify/GitHub pages) as demos.

What I've found helped me solidify my learning is documenting what
I've done (like the build an MDX blog). You don't have to share it on
dev.to or Twitter but you can link to it for examples of how you
worked through something if any potential employers ask.

Please don't take this as you _have_ to have all these done, they
help, that's it, please don't feel obliged to do all this, there's
enough unnecessary pressure on everyone at the moment so the last
thing I want to do is add to yours.

## The long one!

I'm going to go though my experience of getting my first job as a web
developer back in March 2018.

So I'm comming up on four years as a web dev 😬, prior to that I
worked as an analyst developer for large financial organisations, so I
was familiar with developing, only not in the web space, I did all my
development in Excel, Access and SQL Server.

My path to web development started in the summer of 2016 I first tried
to do web development in Python with Django and Flask(??) I found
these super confusing and it wasn't until I saw a tweet and
accompanying post from Quincy Larson saying "bet on JavaScript". From
that I discovered freeCodeCamp and it opened my eyes to how large a
'thing' I was getting into here.

I worked my way though the freeCodeCamp curriculum whist also doing my
day job as a contractor, I'd have to get up a couple of hours before
work to get some coding in before going to work on spreadsheets all
day then an hour or so in the evenings.

I also started the #100DaysOfCode challenge toward the end of 2016
this is great for helping get used to coding an hour each day and in
my case has tuned into a bit of a bad habit as I still get up super
early and code. **I'm not saying you have to do this to be successful,
this is what I have done** these are my experiences and they will
differ from yours, I'm sharing what I did.

It wasn't until I did a build to learn project with several other
cohorts at the end of 2017 in the [Chingu] collabs that I felt
confident enough to start applying for jobs. **Plot twist** I could
have been applying way before then and the only person stopping me was
me 😊

The build to learn projects are a good indication of how you would
work in a team of other developers as well.

For me I was super lucky to find a company that wanted my previous
experience in the financial sector and working in large organisations.

Again I'll stress I was super lucky.



[my newsletter]: https://ss10.dev/nl
[chingu]: https://chingu.io/

**Comment by nicholasrokosz** [comment]
Scott,
Thanks for the great response! (and stickers!)
I really like the idea of writing online to solidify and demonstrate what I'm learning. I'm working on adding a blog/garden/log to my site now.
Also rising early to write code before work seems like a good habit!

Best,
Nick
